Display inbox count in the WP Dashboard

saifsultanc opened the pull request:
## Description
Added the red bubble inbox count display with the Workflow Label on the WP Dashboard

## Testing instructions
Switch to this branch and confirm the presence of the 'red bubble' with the inbox entries count next to the 'Workflow' on the WP Dashboard.

## Automated Test Enhancements
N/A
 
## Screenshots
N/A

## Documentation Changes?
N/A

## Checklist:
- [x] I've tested the code.
- [x] My code follows the WordPress code style. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code follows the inline documentation standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/ -->

stevehenty commented on class-gravity-flow.php:
```diff
@@ -218,6 +218,8 @@ public function init() {
 				)
 			);
 
+			add_filter('add_menu_classes', array( $this, 'show_notification_count' ), 10);
```
```suggestion
			add_filter( 'add_menu_classes', array( $this, 'show_notification_count' ), 10);
```

stevehenty commented on class-gravity-flow.php:
```diff
@@ -5957,6 +5959,25 @@ public function filter_after_update_entry( $form, $entry_id ) {
 
+		/**
+		 * Add inbox notification count to Workflow Menu.
+		 *
```
missing since tag

stevehenty commented on class-gravity-flow.php:
```diff
@@ -5957,6 +5959,25 @@ public function filter_after_update_entry( $form, $entry_id ) {
+		public function show_notification_count( $menu ) {
+			$workflow_menu_pos = -1;
+			foreach ( $GLOBALS['menu'] as $menuitem ) {
+				if( $menuitem[0] == "Workflow" ) {
```
```suggestion
				if ( $menuitem[0] == "Workflow" ) {
```

stevehenty commented on class-gravity-flow.php:
```diff
@@ -5957,6 +5959,25 @@ public function filter_after_update_entry( $form, $entry_id ) {
+		 */		
+		public function show_notification_count( $menu ) {
+			$workflow_menu_pos = -1;
+			foreach ( $GLOBALS['menu'] as $menuitem ) {
```
Let's use `global $menu` at the top instead of using $GLOBALS.

stevehenty commented on class-gravity-flow.php:
```diff
@@ -5957,6 +5959,25 @@ public function filter_after_update_entry( $form, $entry_id ) {
+			}
+
+			$pending_count = $this->get_workflow_count();
+			$menu[$workflow_menu_pos][0] = sprintf( __('Workflow %s'), "<span class='update-plugins count-$pending_count'><span class='plugin-count'>" . number_format_i18n($pending_count) . "</span></span>" );
```
```suggestion
			$menu[ $workflow_menu_pos ][0] = sprintf( __( 'Workflow %s' ), "<span class='update-plugins count-$pending_count'><span class='plugin-count'>" . number_format_i18n($pending_count) . "</span></span>" );
```

stevehenty replied:
Please check that this also works after changing the local of the site.

stevehenty commented on class-gravity-flow.php:
```diff
@@ -6506,6 +6527,23 @@ public function get_user_roles( $user_id = null ) {
+		 */		
+		public function get_workflow_count() {
+			$count_value = get_transient( 'gflow_inbox_count' );
+			if( $count_value === false) {
```
```suggestion
			if ( $count_value === false) {
```

stevehenty commented on class-gravity-flow.php:
```diff
@@ -6506,6 +6527,23 @@ public function get_user_roles( $user_id = null ) {
+			$count_value = get_transient( 'gflow_inbox_count' );
+			if( $count_value === false) {
+				$count_value = Gravity_Flow_API::get_inbox_entries_count();
+				set_transient( 'gflow_inbox_count', $count_value, MINUTE_IN_SECONDS );
```
This will impact all users on the site.

saifsultanc commented:
> This is working but only if there's only one user on the site using the inbox. We'll need to handle multiple users. I'm also seeing some errors when the "Workflow" label is customised in the label settings.
> `PHP Notice: Undefined offset: 2 in /app/public/wp-admin/menu-header.php on line 91`

Fixed the custom label settings issue. Issues for multiple users also addressed, along with the formatting fixes.

stevehenty commented on change_log.txt:
```diff
@@ -1 +1,2 @@
 - Fixed an issue where GP Limit Dates does not function on the User Input step when the min and max range is based on non-editable Date fields.
+- Added the red bubble inbox count display on the WP Dashboard Workflow display.
```
```suggestion
- Added the red bubble inbox count display on the WP Dashboard Workflow menu item.
```

stevehenty commented on class-gravity-flow.php:
```diff
@@ -6506,6 +6534,23 @@ public function get_user_roles( $user_id = null ) {
+		 * @return int
+		 */		
+		public function get_workflow_count() {
+			$count_value = get_transient( 'gflow_inbox_count' . get_current_user_id()  );
```
```suggestion
			$count_value = get_transient( 'gflow_inbox_count_' . get_current_user_id()  );
```

stevehenty commented on class-gravity-flow.php:
```diff
@@ -6506,6 +6534,23 @@ public function get_user_roles( $user_id = null ) {
+		 * 
+		 * @return int
+		 */		
+		public function get_workflow_count() {
```
Would this make more sense?
```suggestion
		public function get_inbox_count() {
```

stevehenty commented on class-gravity-flow.php:
```diff
@@ -5957,6 +5959,32 @@ public function filter_after_update_entry( $form, $entry_id ) {
+		 * 
+		 * @param array $menu The current WP Dashboard Menu.
+		 */		
+		public function show_notification_count( $menu ) {
```
Would this be better?
```suggestion
		public function show_inbox_count( $menu ) {
```
